Textarea style

johangiraud opened the pull request.

johangiraud committed:
tiny little changes on editing place to make author <select> 100% width

johangiraud committed:
make textarea auto-expand when typing or pasting content.

johangiraud committed:
Merge branch 'master' into textarea-style

johangiraud committed:
auto-expand textarea when removing text + keep height when saving article

johangiraud committed:
Merge branch 'textarea-style' of github.com:cafebabel/cafebabel.com into textarea-style

davidbgk commented on cafebabel/templates/articles/edit.html:
```diff
@@ -67,7 +67,7 @@ <h1 class="edit {% if article.id %}active{% endif %}">
   </div>
   <div class=textarea>
-    <label for=summary>Summary</label>
+    <label for=summary>Teaser</label>
```
What about translations and tags which have the same label?

johangiraud replied:
Translations won't have the same _Teaser_. Never. And for the tags, I think we said we don't care if tags are spelled the same. MeetMtHood in english won't have nothing to do with MeetMyHood in french. They both exists and have a list of articles, that's all. Right ?

davidbgk replied:
For tags I get it but for translations you will keep Summary within the form and it will be inconsistent with that one (Teaser)

johangiraud replied:
should I change the `for=summary` to `for=teaser`?

vinyll replied:
I suppose that's a good question that will occur in multiple places. We might need a namespace that would make a context. But that will be when translating I suppose.

davidbgk replied:
@cafebabel-webmaster no, I mean [here](https://github.com/cafebabel/cafebabel.com/blob/master/cafebabel/templates/articles/translations/edit.html#L17) and [there](https://github.com/cafebabel/cafebabel.com/blob/master/cafebabel/templates/articles/translations/create.html#L19) it will create inconsistency in the naming.

vinyll replied:
Ok, in other word here for an article we call it "Teaser", but [for a translation we call it "Summary"](https://github.com/cafebabel/cafebabel.com/blob/master/cafebabel/templates/articles/translations/edit.html#L17).
The question is why do we call it different here, or if they all should be called "Teaser", or this one should remain "Summary".

johangiraud replied:
done: https://github.com/cafebabel/cafebabel.com/pull/220/commits/ee76d5d98f6eee4f2af284c9157c206fad47b4a3, https://github.com/cafebabel/cafebabel.com/pull/220/commits/1156dab3d50e84989a5bb2bb13f1975655722fb2

davidbgk commented on cafebabel/static/js/script.js:
```diff
@@ -113,3 +113,38 @@ Array.from(document.querySelectorAll('h1.edit input')).forEach(inputh1 => {
+
+/* Auto-expand textarea when typing http://jsfiddle.net/hmelenok/WM6Gq/ */
+var textAreaAutoResize
+if (window.attachEvent) {
```
Is that test consistent with other scripts?

johangiraud replied:
It works, that's all. I guess it could be cleaner. It that what you mean ?

vinyll replied:
Right, I think we only use `addEventListener` and we don't make backward compatibility to IE5/8

johangiraud replied:
Ah sorry I got what you mean. It could be in a separate script. A script only called on article edit pages

davidbgk replied:
No, I mean we did not take care of that retro-compatibility since now.

johangiraud replied:
Ok you'll explain that later

vinyll replied:
@cafebabel-webmaster https://stackoverflow.com/a/30915025/328117.
We don't treat older browsers cases.

johangiraud replied:
ok

vinyll commented on cafebabel/static/js/script.js:
```diff
@@ -113,3 +113,38 @@ Array.from(document.querySelectorAll('h1.edit input')).forEach(inputh1 => {
+    element.attachEvent('on' + event, handler)
+  }
+} else {
+  textAreaAutoResize = function(element, event, handler) {
```
As done in other places, `function textAreaAutoresize(…` would be prefered and you could also get rid of `var textAreaAutoResize`

vinyll commented on cafebabel/static/js/script.js:
```diff
@@ -113,3 +113,38 @@ Array.from(document.querySelectorAll('h1.edit input')).forEach(inputh1 => {
+function textAreaResizing() {
+  Array.from(document.querySelectorAll('textarea')).forEach(textArea => {
+    function resize() {
+      textArea.style.height = 'auto'
```
Why do we first set it to `'auto'` and then immediately change it?

johangiraud replied:
I don't know.

vinyll replied:
Can we remove it then?

johangiraud replied:
ok

vinyll commented on cafebabel/static/js/script.js:
```diff
@@ -113,3 +113,38 @@ Array.from(document.querySelectorAll('h1.edit input')).forEach(inputh1 => {
+}
+function textAreaResizing() {
+  Array.from(document.querySelectorAll('textarea')).forEach(textArea => {
+    function resize() {
```
Not certain why we need to declare this function inside of the forEach 🤔

johangiraud replied:
I worked only on the first `<textarea>` otherwise

vinyll replied:
yes, but if you declare the function outside and you pass the `textArea` as an argument.
I suppose there was a reason but it looks very unsusual. Maybe it has to do with the setTimeout. (?)

johangiraud replied:
I can't get it work else way :-(

johangiraud committed:
refactor no need older browser support

vinyll commented on cafebabel/static/js/script.js:
```diff
@@ -116,15 +116,11 @@ Array.from(document.querySelectorAll('h1.edit input')).forEach(inputh1 => {
-    element.addEventListener(event, handler, false)
-  }
+
+textAreaAutoResize = (element, event, handler) => {
```
this should be a `function textAreaAutoResize…` and remove the `var textAreaAutoResize`

johangiraud replied:
ok

johangiraud committed:
Merge branch 'master' into textarea-style

johangiraud committed:
clean up auto-expand textarea functions

johangiraud committed:
minor style changes on article text and editing place text

johangiraud committed:
add Participate & Translate buttons

johangiraud committed:
change preloader name to avoid safari issue

johangiraud committed:
add englsih version social network links hardcoded

johangiraud committed:
changin profile picture recommandation sentence style

johangiraud committed:
change px to rem unit + menu item in the foreground

johangiraud committed:
minor layout changes in translation/detail

johangiraud committed:
minor titles styles in articles

johangiraud committed:
remove useless pitch an idea page styles

vinyll commented on cafebabel/static/js/script.js:
```diff
@@ -113,3 +113,31 @@ Array.from(document.querySelectorAll('h1.edit input')).forEach(inputh1 => {
   )
 })
+
+/* Auto-expand textarea when typing http://jsfiddle.net/hmelenok/WM6Gq/ */
```
This whole script being an external library, the functions it should be in an isolated file and then called here like you did on line 143.

johangiraud replied:
ok done. https://github.com/cafebabel/cafebabel.com/pull/220/commits/d149accdc65015f66500169fea80accaed71772f

vinyll commented on cafebabel/templates/base.html:
```diff
@@ -56,12 +56,15 @@
   </ul>
 </nav>
 
+<button id=participate><a href=#>Participate</a></button>
```
Interactive element (`a`) inside of an interactive element (`button`) looks confusing.
It should either be a link to a different content (`a`), or be an interactive element (`button`).

`a href=#` makes me think it's not an actual link…

vinyll commented on cafebabel/templates/base.html:
```diff
@@ -56,12 +56,15 @@
-    <li><a href=# class=icon-twitter></a></li>
-    <li><a href=# class=icon-instagram></a></li>
-    <li><a href=# class=icon-youtube></a></li>
+    <li><a href=https://www.facebook.com/cafebabelenglish class=icon-facebook></a></li>
```
:+1: This will permit to test properly and know what the links look like.

vinyll reviewed (changes requested):
Sorry, I meant to approve the last commit and I didn't realize it was mixed into this larger PR.

johangiraud committed:
textarea auto-expand in a separate file

johangiraud committed:
change button > a to a

johangiraud committed:
add links to Participate and Translate a

johangiraud committed:
mini style change on label after translate page

johangiraud committed:
Merge branch 'master' into textarea-style

johangiraud committed:
change forgotten font path

johangiraud committed:
social networks icons in main menu on mobile

vinyll commented:
I suppose it's way too late now, but this PR is suppose to be about textarea styles.
It seems like a lot has got nothing to do with the original point of the PR.

It's very fine as long as we are all ok that it will therefore take more time to review and to merge as one blockage will block it all. ;)

johangiraud commented:
@vinyll you are totally right. I find sometimes easier to do minor changes in the same PR instead of reopening another one. But that's WRONG !!!

vinyll commented on cafebabel/static/js/textarea-change.js:
```diff
@@ -0,0 +1,27 @@
+  })
+}
+
+textAreaResizing()
```
I didn't see this one line should be here but I guess it make sense too :)

johangiraud replied:
well since `textAreaResizing` is a function, it doesn't call it itself, does it? So we have to call it at one point

davidbgk replied:
A common pattern in JS is to use an [IIFE](https://en.wikipedia.org/wiki/Immediately-invoked_function_expression)

vinyll replied:
Most of the purpose of IIFEs may have been replaced with the local scope of ES6 (that's one reason of using `const`).
Therefore we could have the code directly outside of a function but I guess it it more readable this way.
I had in mind to call `textAreaResizing()` from the main _scripts.js_ file but it doesn't feel ackward to me as such.
Any thought?

davidbgk replied:
It would mean that `textarea-change.js` should be included everywhere `scripts.js` is in use?

johangiraud replied:
I don't think we need `texarea-change.js` each time we need `script.js`. I am even pretty sure of it.

vinyll reviewed (commented):
[This point](https://github.com/cafebabel/cafebabel.com/pull/220/files#r166966400) remains pending.

johangiraud committed:
change 'Summary' for 'Teaser'

johangiraud commented:
About the _summary_ issue, I have replaced to one left and changed it to _Teaser_. Anyway since the beginning, you guys have chosen the words in use. When I thought we could translate the website in 5 languages, I planned to go through the whole _wording_ issue is the editors when we would have translated the website. But since we can't do this anymore, I guess, we have to do that for the english version very soon with Lara (english editor).

johangiraud committed:
change 'Summary' for 'Teaser' 2

davidbgk commented on cafebabel/static/js/textarea-change.js:
```diff
@@ -0,0 +1,27 @@
+/* Auto-expand textarea when typing http://jsfiddle.net/hmelenok/WM6Gq/ */
+function textAreaAutoResize(element, event, handler) {
```
I don't get why we need that function instead of directly calling `addEventListener` below?

vinyll replied:
OH, I think it's because in the copied version there was a switch between `addEventListener` and `attachEvent`. I guess there's no more reason now.

johangiraud replied:
Feel free to edit this yourself please @vinyll @davidbgk . I don't know the difference between js versions. Just make it consistant with the rest we've been doing: ES6, right?

johangiraud replied:
I am sure you are right, and you see in 2 sec if we need this function or if we don't

davidbgk commented on cafebabel/templates/articles/translations/detail.html:
```diff
@@ -7,26 +7,26 @@
 <article>
-  <h1>{{ translation.title }}</h1>
-
-  <p class=reading-time>Reading time: {{ translation.body|reading_time }} min.</p>
```
No more reading time for translations?

johangiraud replied:
no..

johangiraud committed:
Merge branch 'master' into textarea-style

johangiraud committed:
cleaning out forgotten useless stuff

johangiraud committed:
Merge branch 'textarea-style' of github.com:cafebabel/cafebabel.com into textarea-style

davidbgk committed:
Simplify auto-expand textarea

vinyll reviewed (approved):
It seems fine to me

johangiraud commented:
great !
